app/controllers/incomeSources/add/BusinessAccountingMethodController.scala

```diff
@@ -57,22 +57,23 @@ class BusinessAccountingMethodController @Inject()(val authenticate: Authenticat
                    (implicit user: MtdItUser[_], hc: HeaderCarrier, ec: ExecutionContext, messages: Messages): Future[Result] = {
 
     val incomeSourcesEnabled: Boolean = isEnabled(IncomeSources)
-    val backUrl: String = if (isAgent) controllers.incomeSources.add.routes.CheckBusinessDetailsController.showAgent().url else
-      controllers.incomeSources.add.routes.CheckBusinessDetailsController.show().url
+    val backUrl: String = if (isAgent) controllers.routes.AddBusinessAddressController.showAgent().url else
+      controllers.routes.AddBusinessAddressController.show().url
     val postAction: Call = if (isAgent) controllers.incomeSources.add.routes.BusinessAccountingMethodController.submitAgent() else
       controllers.incomeSources.add.routes.BusinessAccountingMethodController.submit()
 
     val errorHandler: ShowInternalServerError = if (isAgent) itvcErrorHandlerAgent else itvcErrorHandler
 
     if (incomeSourcesEnabled) {
       val userBusinesses: List[BusinessDetailsModel] = user.incomeSources.businesses
-      if (shouldAutomaticallyRedirect(userBusinesses)) {
+      if (user.incomeSources.checkBusinessesCashOrAccrualsTypes && !userBusinesses.forall(_.isCeased)) {
+        val accountingMethod: String = userBusinesses.head.cashOrAccruals.getOrElse(throw MissingFieldException("cashOrAccruals field missing"))
```

Comment:
* lines 69 - 70 can be moved within relevant model and we are not handling error which can be raised here